src/web/user-summary.jsp

```diff
@@ -48,23 +48,69 @@
     </head>
     <body>
 
-<%  // Get parameters
-    int start = ParamUtils.getIntParameter(request,"start",0);
-    int range = ParamUtils.getIntParameter(request,"range",webManager.getRowsPerPage("user-summary", DEFAULT_RANGE));
+<%
+    // Get the presence manager
+    PresenceManager presenceManager = webManager.getPresenceManager();
 
-    if (request.getParameter("range") != null) {
-        webManager.setRowsPerPage("user-summary", range);
+    // By default, display all users
+    Predicate<User> filter = new Predicate<User>() {
+        @Override
+        public boolean test(final User user) {
+            return true;
+        }
+    };
+    final String searchUsername = ParamUtils.getStringParameter(request, "searchUsername", "");
+    if (!searchUsername.trim().isEmpty()) {
+        final String searchCriteria = searchUsername.trim().toLowerCase();
+        filter = filter.and(new Predicate<User>() {
+            @Override
+            public boolean test(final User user) {
+                return user.getUsername().toLowerCase().contains(searchCriteria);
+            }
+        });
+    }
+    final String searchName = ParamUtils.getStringParameter(request, "searchName", "");
+    if (!searchName.trim().isEmpty()) {
+        final String searchCriteria = searchName.trim().toLowerCase();
+        filter = filter.and(new Predicate<User>() {
+            @Override
+            public boolean test(final User user) {
+                return user.getName().toLowerCase().contains(searchCriteria);
+            }
+        });
+    }
+    final String searchGroup = ParamUtils.getStringParameter(request, "searchGroup", "");
+    final GroupManager groupManager = webManager.getGroupManager();
+    if (!searchGroup.isEmpty()) {
+        final Predicate<String> searchPredicate = new Predicate<String>() {
+            @Override
+            public boolean test(final String groupName) {
+                return searchGroup.equals(groupName);
+            }
+        };
+        filter = filter.and(new Predicate<User>() {
+            @Override
+            public boolean test(final User user) {
+                return groupManager.getGroups(user).stream()
+                    .map(new Function<Group, String>() {
+                        @Override
+                        public String apply(final Group group) {
+                            return group.getName();
+                        }
+                    })
+                    .anyMatch(searchPredicate);
+            }
+        });
     }
 
-    // Get the user manager
-    int userCount = webManager.getUserManager().getUserCount();
-
-    // Get the presence manager
-    PresenceManager presenceManager = webManager.getPresenceManager();
+    final ListPager<User> listPager = new ListPager<>(request, response, new ArrayList<>(webManager.getUserManager().getUsers()), filter,
```

Comment:
This, sadly, won't work on a server with lots of users. Getting them all from database (and keeping them in memory) is simply to much of a resource hog. Ignite Realtime (250,000 users, last I checked) has been running with 100% CPU since it started, 40 minutes ago. I believe this to be the culprit.

Reply:
Some comments added to https://issues.igniterealtime.org/browse/OF-1542 for a new approach - any further suggestions appreciated. 

Though 250,000 is not a big number today. If the core user details (username, name, email) are ~100 bytes, that's less than 25MB. The real issue is that this 25MB is fetched over I/O each and every time user-summary is accessed with the original implementation. I would guess that it was necessary to cache only recently accessed users some years back when less RAM was generally available.